src/IronRure/Options.cs

```diff
@@ -1,38 +1,61 @@
 using System;
 
-namespace IronRure
+namespace IronRure;
+
+/// <summary>
+/// Regex compilation options.
+/// </summary>
+public sealed class Options : IDisposable
 {
-    /// <summary>Regex compilation options.</summary>
-    public class Options : IDisposable
+    private bool _disposed;
+
+    /// <summary>
+    /// Creates a new options instance with the default values.
+    /// </summary>
+    public Options()
     {
-        /// <summary>Create a new options instnace with the default values.</summary>
-        public Options()
-        {
-            Raw = RureFfi.rure_options_new();
-        }
+        Raw = RureFfi.rure_options_new();
+    }
 
-        internal OptionsHandle Raw { get; }
+    internal OptionsHandle Raw { get; }
 
-        /// <summary>
-        ///   Set Size Limit - Controls the size of a single regex program
-        /// </summary>
-        public uint Size
+    /// <summary>
+    /// Sets the size limit for a single regex program.
+    /// </summary>
+    public uint Size
+    {
+        set
         {
-            set => RureFfi.rure_options_size_limit(Raw, new UIntPtr(value));
+            if (_disposed)
+                throw new ObjectDisposedException(nameof(Options));
+            RureFfi.rure_options_size_limit(Raw, new UIntPtr(value));
         }
+    }
 
-        /// <summary>
-        ///   Set DFA Size Limit - Controls the DFA cache size during search
-        /// </summary>
-        public uint DfaSize
+    /// <summary>
+    /// Sets the DFA size limit for the DFA cache during search.
+    /// </summary>
+    public uint DfaSize
+    {
+        set
         {
-            set => RureFfi.rure_options_dfa_size_limit(Raw, new UIntPtr(value));
+            if (_disposed)
+                throw new ObjectDisposedException(nameof(Options));
+            RureFfi.rure_options_dfa_size_limit(Raw, new UIntPtr(value));
         }
+    }
 
-        /// <inheritdoc />
-        public void Dispose()
+    /// <summary>
+    /// Releases all resources used by this instance.
+    /// </summary>
+    public void Dispose()
+    {
+        if (_disposed)
         {
-            Raw.Dispose();
+            return;
         }
+
+        Raw?.Dispose();
```

Comment:
I don’t think the handles can ever be null in these dispose calls can they?

Reply:
In regards to this, Im not sure, if I went all the way down and enabled nullable it would suggest so (which I tried and gave up on cause It required me to think too much about nullability, the source library and how we want to handle null, this is technically just a leftover off that, so I have reverted it.